ground/src/test/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerFragmentTest.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.offlineareas.viewer
+
+import android.os.Bundle
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.hasDescendant
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.isEnabled
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import com.google.android.ground.BaseHiltTest
+import com.google.android.ground.R
+import com.google.android.ground.launchFragmentWithNavController
+import com.google.android.ground.persistence.local.stores.LocalOfflineAreaStore
+import com.sharedtest.FakeData.OFFLINE_AREA
+import dagger.hilt.android.testing.HiltAndroidTest
+import javax.inject.Inject
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner::class)
+class OfflineAreaViewerFragmentTest : BaseHiltTest() {
```

Comment:
Can we add tests where there are no offline areas in local db?

Reply:
done

Reply:
Why do we need a different class for this? We can move the setup() call to the test itself so that each test can configure the fragment accordingly.

Reply:
Tried, but one of the test fails, and interesting part is, it fails only when. you run the whole class test, all test together, but when running alone it passes.

Thinking for a solution, as it look like it have to do something with the testDispatcher keeping itself to Main Thread and not resetting 

![Screenshot 2024-08-14 at 9 18 29 PM](https://github.com/user-attachments/assets/62d63a46-3871-4f9f-a3bb-49924c32caed)

Reply:
I tried to run them in order using `@FixMethodOrder(MethodSorters.NAME_ASCENDING)` and all test case passed

![Screenshot 2024-08-14 at 9 49 38 PM](https://github.com/user-attachments/assets/7d2bea44-f8d8-43d3-9a62-630dc0e1c2e3)